tests/worker/test_workflow.py

```diff
@@ -274,6 +300,43 @@ async def test_workflow_signal_and_query_handlers(client: Client):
         )
 
 
+@workflow.defn
+class SignalAndQueryErrorsWorkflow:
+    @workflow.run
+    async def run(self) -> None:
+        # Wait forever
+        await asyncio.Future()
+
+    @workflow.signal
+    def bad_signal(self) -> NoReturn:
+        raise ApplicationError("signal fail", 123)
+
+    @workflow.query
+    def bad_query(self) -> NoReturn:
+        raise ApplicationError("query fail", 456)
```

Comment:
nit: don't need an ApplicationError here

Reply:
We do if I want to test how details are handled :-) Though I note "(no details on query failure)" in the test, so it became useless.